lib/web_ui/test/ui/vertices_test.dart

```diff
@@ -58,6 +65,120 @@ void testMain() {
     await drawPictureUsingCurrentRenderer(recorder.endRecording());
     await matchGoldenFile('ui_vertices_antialiased.png', region: region);
   }, skip: isHtml); // https://github.com/flutter/flutter/issues/127454
+
+  test('Vertices assert checks', () {
+    // We don't test textureCoordinate assert checks on html render because HTML renderer's SurfaceVertices() does not support textureCoordinates
+    if (renderer.rendererTag != 'html') {
+      try {
+        ui.Vertices(
+          ui.VertexMode.triangles,
+          const <ui.Offset>[ui.Offset.zero, ui.Offset.zero, ui.Offset.zero],
+          textureCoordinates: const <ui.Offset>[ui.Offset.zero],
+        );
+        if (assertsEnabled) {
+          throw AssertionError('Vertices did not throw the expected assert error.');
+        }
```

Comment:
for example, in an else block here call `Canvas.drawVertices(vertices, etc.)`

Reply:
@dnfield I have added a function which uses and disposes of the created invalid `ui.Vertices` objects when assertions are not enabled.  Each of the various created (when assertions are disabled) invalid Vertices object's use will be tested this way.  In this way we will answer your question to determine "segfault vs. safe to use".

(additionally, within `testing\dart\painting_test.dart`,  I added calls to dispose() of the invalid `Vertices` objects that are created if assertions are not enabled (but here I did not attempt to use them).

```dart

// This function is used when asserts are NOT enabled and vertices() allows invalid 
// vertices objects to be created.  Here we attempt to use these invalid ui.Vertices
// objects to determine if they cause problems within drawVertices()
void useAndDisposeOfInvalidVerticesObject( ui.Vertices vertices ) {
  expect(vertices.debugDisposed, isFalse);

  final ui.PictureRecorder recorder = ui.PictureRecorder();
  final ui.Canvas canvas = ui.Canvas(
    recorder,
    const ui.Rect.fromLTRB(0, 0, 100, 100)
  );
  canvas.drawVertices(
    vertices,
    ui.BlendMode.srcOver,
    ui.Paint(),
  );
  vertices.dispose();
  expect(vertices.debugDisposed, isTrue);
}
```

Used like this throughout `vertices_test.dart` for all created/invalid `ui.Vertices` objects when assertions are not enabled:
```dart
try {
        final ui.Vertices invalidVertices = ui.Vertices(
          ui.VertexMode.triangles,
          const <ui.Offset>[ui.Offset.zero, ui.Offset.zero, ui.Offset.zero],
          textureCoordinates: const <ui.Offset>[ui.Offset.zero],
        );
        if (assertsEnabled) {
          throw AssertionError('Vertices did not throw the expected assert error.');
        } else {
          // Assertions are NOT ENABLED, so we will attempt to use this invalid ui.Vertices object
          // we just created to check if that causes exceptions/problems.
          useAndDisposeOfInvalidVerticesObject(invalidVertices);
        }
      } on AssertionError catch (e) {
        expect('$e', contains(r'\"positions\" and \"textureCoordinates\" lengths must match.'));
      }
```